learnEcl/2500-transform.md

```diff
@@ -2,3 +2,196 @@
+Transform can be used with PROJECT, JOIN, ITERATE, ROLLUP and more.
+
+## Syntax
+
+<pre>
+<EclCode code="EXPORT [return_dataset_layout] transform_name ([input_arguments_types]+ arg_name ) := TRANSFORM
+      SELF.return_field_name := arg_name.input_dataset_fieldname;
+      SELF := arg_name;
+      SELF := [];
+END;">
+</EclCode>
+</pre>
+
+| _Value_ | _Definition_ |
+| :- | :- |
+| EXPORT | Optional. Used in MODULEs |
+| return_dataset_layout | Record-definition/layout of result dataset. |
+| transform_name | The name by which the transform will be invoked. |
+| input_arguments_types | The argument’s data type. If passing a dataset, the data type is DATASET(record_definition). |
+| arg_name | Used to reference your argument in the transform. |
+| TRANSFORM | Required. |
+| SELF | Reference the field in the return_data_type. |
+| return_field_name | Refers to the field in result dataset. |
+| input_Dataset_fieldname |	Refers to the field in the input dataset. |
+| SELF := [ ] | Assign default value for every field in result dataset that doesn't have a defined operation or doesn't exists in the input dataset. For example if there is a INTEGER field in the result dataset, that transform didn't assign a definition to it, the field will receive a 0 which is the default value for INTEGER. |
+| END | Required. |
+
+## Transform Type One (Standalone TRANSFORM)
+
+If you need the transform to be used in multiple places, or it contains many fields or child datasets, you may want to define a standalone transform (a function that can be called multiple times)
+
+### Demo Dataset
+
+| FirstName | LastName |
+| :- | :- |
+| Sun | Shine |
+| Blue | Moon |
+| Silver | Rose |
+
+**Example**
+
+<pre>
+<EclCode
+id="TransformExp_1"
+tryMe="TransformExp_1"
+code="/*Transform Example:*/
+
+/*
+TRANSFORM Example: 
+Using an input dataset, to concat names and count the number of rows in the input dataset. 
+PROJECT result always have the same number of rows as input dataset. 
+*/
+
+
+// Defining record layout
+Names_layout := RECORD 
+    STRING FirstName;
+    STRING LastName;
+END;
+
+// Creating Explicit dataset
+Names_DS := DATASET([
+              {'Sun','Shine'},
+              {'Blue','Moon'},
+              {'Silver','Rose'}],
+              Names_layout);
+
+// Defining new layout for the project result
+NameOutRec := RECORD
+    STRING  FirstName;
+    STRING  LastName;
+    STRING  CatValues;
+    INTEGER RecCount; //Counter
+END;
+
+/*
+NameOutRec: Result of the project gets saved in this record layout
+CatThem: TRANSFORM name
+Names_layout L: Left datasets that’s passed through project
+INTEGER C: Counter
+*/
+NameOutRec CatThem(Names_layout L, INTEGER C) := TRANSFORM
+
+  // Contacting FirstName with LastName and adding space between them
+  SELF.CatValues := L.FirstName + ' ' + L.LastName; 
+  SELF.RecCount := C; // Counter
+  // Assign default values to all fields that are in result dataset and haven't been define in this TRANSFORM
+  SELF := L; 
```

Comment:
This does not match the comment just before it.  This syntax:

SELF := L

performs a name-wise copy of fields that have not already been assigned, using 'L' as a source.